src/controllers/api/monsterController.js

```diff
@@ -4,14 +4,51 @@ const utility = require('./utility');
 const { redisClient } = require('../../util');
 const getAsync = promisify(redisClient.get).bind(redisClient);
 
-exports.index = async (req, res, next) => {
-  const search_queries = {};
-  if (req.query.name !== undefined) {
-    search_queries.name = { $regex: new RegExp(utility.escapeRegExp(req.query.name), 'i') };
+/**
+ * Builds a MongoDB query based on provided URL query params in the request.
+ * Used to filter entries to those matching particular values
+ * @param queryParams URL query params in the API request
+ */
+function buildQuery(queryParams) {
+  const query = {};
+
+  if (queryParams.name) {
+    query.name = { $regex: new RegExp(utility.escapeRegExp(queryParams.name), 'i') };
+  }
+  if (queryParams.challenge_rating) {
+    query.challenge_rating = { $in: queryParams.challenge_rating.split(',') };
+  }
+  if (queryParams.alignment) {
+    query.alignment = { $regex: utility.csvToIRegExp(queryParams.alignment) };
+  }
+  if (queryParams.type) {
```

Comment:
You may as well through in `subtype` for good measure.

Reply:
Also, all of this feels like a really good reason for me to finish setting up GQL fully instead.